google/cloud/pubsub/internal/noop_message_batch.h

```diff
@@ -33,9 +34,26 @@ class NoOpMessageBatch : public MessageBatch {
 
   void SaveMessage(pubsub::Message) override{};
 
-  void Flush() override{};
+  future<void> Flush() override {
+    future<void> result;
+    {
+      std::lock_guard<std::mutex> lk(mu_);
+      waiter_.get_future();
+    }
+    return result;
```

Comment:
I meant to say

```
    future<void> result;
    {
      std::lock_guard<std::mutex> lk(mu_);
      result = waiter_.get_future();
    }
    return result;
``` 

So that the promise could be satisfied in the FlushCallback.

But now this irrelevant.